src/App.tsx

```diff
@@ -1,17 +1,38 @@
 import { BrowserRouter, Routes, Route } from 'react-router-dom'
 import LoginPage from '@/pages/login'
-import Home from '@/pages/home'
+import HomePage from '@/pages/HomePage'
 import NotFound from '@/pages/NotFound'
+import MyPage from '@/pages/MyPage'
+import OrderPage from '@/pages/OrderPage'
+import { AuthProvider } from '@/contexts/AuthContext'
+import ProtectedRoute from '@/components/ProtectedRoute'
+import PageWrapper from './components/PageWrapper'
 
 function App() {
   return (
-    <BrowserRouter>
-      <Routes>
-        <Route path="/login" element={<LoginPage />} />
-        <Route path="/" element={<Home />} />
-        <Route path="*" element={<NotFound />} />
-      </Routes>
-    </BrowserRouter>
+    <AuthProvider>
+      <BrowserRouter>
+        <Routes>
+          <Route path="/login" element={<PageWrapper><LoginPage /></PageWrapper>} />
+          <Route path="/" element={<PageWrapper><HomePage /></PageWrapper>} />
+          <Route path="/my" element={
+            <ProtectedRoute>
+              <PageWrapper>
+                <MyPage />
+              </PageWrapper>
+            </ProtectedRoute>
+          } />
+          <Route path="/order/:productId" element={
+            <ProtectedRoute>
+              <PageWrapper>
+                <OrderPage />
+              </PageWrapper>
+            </ProtectedRoute>
+          } />
+          <Route path="*" element={<PageWrapper><NotFound /></PageWrapper>} />
+        </Routes>
+      </BrowserRouter>
```

Comment:
모든 페이지에 PageWrapper 컴포넌트가 래핑되는 구조를 생각하신 것 같은데요.

그러면 굳이 각 페이지 컴포넌트마다 PageWrapper 컴포넌트를 래핑하기보다는 전체에 한번만 래핑하는게 깔끔할 것 같아요
```suggestion
      <BrowserRouter>
        <PageWrapper>
          <Routes>
              <Route path="/login" element={<LoginPage />} />
              <Route path="/" element={<HomePage />} />
              <Route path="/my" element={
                <ProtectedRoute>
                    <MyPage />
                </ProtectedRoute>
              } />
              <Route path="/order/:productId" element={
                <ProtectedRoute>
                    <OrderPage />
                </ProtectedRoute>
              } />
              <Route path="*" element={<NotFound />} />
          </Routes>
        </PageWrapper>
      </BrowserRouter>
```